src/java/org/apache/cassandra/index/SecondaryIndexManager.java

```diff
@@ -1808,21 +1847,37 @@ public static void shutdownAndWait(long timeout, TimeUnit units) throws Interrup
      * @param indexQueryPlan index query plan used in the read command
      * @param level consistency level of read command
      */
-    public static <E extends Endpoints<E>> E filterForQuery(E liveEndpoints, Keyspace keyspace, Index.QueryPlan indexQueryPlan, ConsistencyLevel level)
+    public static <E extends Endpoints<E>> E filterForQuery(E liveEndpoints,
+                                                            Keyspace keyspace,
+                                                            Index.QueryPlan indexQueryPlan,
+                                                            ConsistencyLevel level,
+                                                            boolean allowFiltering)
     {
-        E queryableEndpoints = liveEndpoints.filter(replica -> {
+        int initial = liveEndpoints.size();
+        Set<InetAddressAndPort> badNodes = MessagingService.instance().endpointsWithVersionBelow(MessagingService.VERSION_DS_11);
+        if (MessagingService.current_version < MessagingService.VERSION_DS_11)
```

Comment:
Michael is about to commit the bump of version and there were some fixes there, I will get his latest version on rebase